tsdb/engine/tsm1/reader.go

```diff
@@ -202,27 +202,27 @@ func NewTSMReader(f *os.File) (*TSMReader, error) {
 }
 
 func (t *TSMReader) applyTombstones() error {
-	// Read any tombstone entries if the exist
-	tombstones, err := t.tombstoner.ReadAll()
-	if err != nil {
-		return fmt.Errorf("init: read tombstones: %v", err)
-	}
-
-	if len(tombstones) == 0 {
-		return nil
-	}
-
 	var cur, prev Tombstone
-	cur = tombstones[0]
-	batch := []string{cur.Key}
-	for i := 1; i < len(tombstones); i++ {
-		cur = tombstones[i]
-		prev = tombstones[i-1]
-		if prev.Min != cur.Min || prev.Max != cur.Max {
+	batch := make([]string, 0, 4096)
```

Comment:
@sebito91 capacity is explicitly specified for performance reasons. It means that the only one allocation is needed, rather than the several that would likely occur if using `append` many times on an empty slice.